atomic_reactor/config.py

```diff
@@ -0,0 +1,568 @@
+    SKIP_KOJI_CHECK_FOR_BASE_IMAGE_KEY = 'skip_koji_check_for_base_image'
+    DEEP_MANIFEST_LIST_INSPECTION_KEY = 'deep_manifest_list_inspection'
+    FAIL_ON_DIGEST_MISMATCH_KEY = 'fail_on_digest_mismatch'
+    SOURCE_CONTAINER_KEY = 'source_container'
+    OPERATOR_MANIFESTS_KEY = 'operator_manifests'
+    IMAGE_SIZE_LIMIT_KEY = 'image_size_limit'
+    BUILDER_CA_BUNDLE_KEY = 'builder_ca_bundle'
+
+
+class ODCSConfig(object):
+    """
+    Configuration for ODCS integration.
+    """
+    def __init__(self, signing_intents, default_signing_intent):
+        self.default_signing_intent = default_signing_intent
+
+        self.signing_intents = []
+        # Signing intents are listed in reverse restrictive order in configuration.
+        # Since the input signing_intents will be modified by inserting a new
+        # key restrictiveness, this deepcopy ensures the original
+        # signing_intent dict objects are not modified accidentally.
+        for restrictiveness, intent in enumerate(reversed(deepcopy(signing_intents))):
+            intent['restrictiveness'] = restrictiveness
+            self.signing_intents.append(intent)
+
+        # Verify default_signing_intent is valid
+        self.get_signing_intent_by_name(self.default_signing_intent)
+
+    def get_signing_intent_by_name(self, name):
+        valid = []
+        for entry in self.signing_intents:
+            this_name = entry['name']
+            if this_name == name:
+                return entry
+
+            valid.append(this_name)
+
+        raise ValueError('unknown signing intent name "{}", valid names: {}'
+                         .format(name, ', '.join(valid)))
+
+    def get_signing_intent_by_keys(self, keys):
+        if isinstance(keys, str):
+            keys = keys.split()
+        keys = set(keys)
+        intents_matching_deprecated_keys = []
+        for entry in reversed(self.signing_intents):
+            keys_set = set(entry['keys'])
+            if (keys and keys_set >= keys) or keys == keys_set:
+                return entry
+
+            permissive_keys_set = set(entry['keys'] + entry.get('deprecated_keys', []))
+            if keys and permissive_keys_set >= keys:
+                intents_matching_deprecated_keys.append(entry)
+
+        if not intents_matching_deprecated_keys:
+            raise ValueError('unknown signing intent keys "{}"'.format(keys))
+
+        logger.warning(
+            'signing intent keys "%s" contain deprecated entries in the "%s" signing intent',
+            keys,
+            intents_matching_deprecated_keys[0]['name']
+         )
+        return intents_matching_deprecated_keys[0]
+
+
+class Configuration(object):
+    """
+    Class to parse the atomic-reactor configuration file
+    """
+    DEFAULT_CONFIG = {ReactorConfigKeys.VERSION_KEY: 1}
+
+    def __init__(self, workflow, config_path=None, basename=REACTOR_CONFIG_BASE_NAME,
+                 env_name=REACTOR_CONFIG_ENV_NAME, raw_config=None):
```

Comment:
Do we need all these ways to init a Config? From a quick look, it seems that only `config_path` and `raw_config` are ever used, and `raw_config` only for unit tests.

How about something like this
```python
class Configuration:
    def __init__(self, workflow, raw_config: dict):
        ...

    @classmethod
    def from_file(cls, workflow, path: str) -> Configuration:
        ...
```
(also, I'm not convinced that `workflow` should be there, see next comment)

Reply:
I did leave reading from env variable be for now

raw_config is just for testing purposes

I prefer to have it as params and call all inits the same way